File: sdk/samples/all in on code/Visual Studio 2008/VBOutlookRibbonXML/ThisAddIn.vb
'************************************* Module Header **************************************'
'* Module Name:	ThisAddIn.vb
'* Project:		VBOutlookRibbonXml
'* Copyright (c) Microsoft Corporation.
'* 
'* The VBOutlookRibbonXml example demonstrates how to use Ribbon XML to create customized 
'* Ribbon for Outlook 2007 inspectors.
'* 
'* This source is subject to the Microsoft Public License.
'* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
'* All other rights reserved.
'* 
'* History:
'* * 9/17/2009 18:00 PM Li ZhenHao Created
'******************************************************************************************


Public Class ThisAddIn

    Private Sub ThisAddIn_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup

    End Sub

    Protected Overrides Function CreateRibbonExtensibilityObject() As Microsoft.Office.Core.IRibbonExtensibility
        ' Outlook will call this method to get our IRibbonExtensibility
        ' implementation.
        Return New Ribbon()
    End Function

    Private Sub ThisAddIn_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown

    End Sub

End Class

File: sdk/samples/all in on code/Visual Studio 2008/CSOutlookRibbonXml/ThisAddIn.cs
/************************************* Module Header **************************************\
* Module Name:  ThisAddIn.cs
* Project:      CSOutlookRibbonXml
* Copyright (c) Microsoft Corporation.
* 
* The CSOutlookRibbonXml example demonstrates how to use Ribbon XML to create customized 
* Ribbon for Outlook 2007 inspectors.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\******************************************************************************************/

#region Using directives
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Linq;
using Outlook = Microsoft.Office.Interop.Outlook;
using Office = Microsoft.Office.Core;
#endregion

namespace CSOutlookRibbonXml
{
    public partial class ThisAddIn
    {
        private void ThisAddIn_Startup(object sender, System.EventArgs e)
        {
        }

        protected override Office.IRibbonExtensibility CreateRibbonExtensibilityObject()
        {
            // Outlook will call this method to get our IRibbonExtensibility
            // implementation.
            return new Ribbon();
        }

        private void ThisAddIn_Shutdown(object sender, System.EventArgs e)
        {
        }

        #region VSTO generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisAddIn_Startup);
            this.Shutdown += new System.EventHandler(ThisAddIn_Shutdown);
        }
        
        #endregion
    }
}
